pkgs/development/compilers/openjdk/8.nix

```diff
@@ -6,6 +6,15 @@
 , minimal ? false
 , enableInfinality ? true # font rendering patch
 , enableGnome2 ? true, gtk2, gnome_vfs, glib, GConf
+, update ? "122", build ? "04"
+, jdk8sha ? "1zqqy5gzrx7f438j5pjdavj41plb04p6b1ikspksrgnhs5wrrr02"
+, langtoolssha ? "0hhsm23mxvjxmf0jxlhm57s203k88s8xbmk71l8zlnjsz88ni4gx"
+, hotspotsha ? "1r4a52brsg1xd2dc2b8lzd4w4yvcjdmj9a6avjihx1hpgcs4xzd1"
+, corbasha ? "0ixa6kdqkiq83817qdymiy772449iva11rh3pr68qpfnmbx1zzil"
+, jdksha ? "1kw4h3j93cvnlzh0vhj4xxdm90bk7hfg6kpqk09x0a12whh2ww3h"
+, jaxwssha ? "0wrj3jyv3922m3pxfg0i9c3ap71b0rass7swvhi996c029rd12r7"
+, jaxpsha ? "0b743mygzdavdd59l98b3l6a03dihs4ipd1xlpkacy778wzpr59d"
+, nashornsha ? "10wkshhzj15wvx7i53dbkwi85f4fbbxi26zphr5b6daf3ib0hind"
```

Comment:
+1 for an attribute set with well known update-build pair and a default value on the latest.

Attribute set could be like : 
~~~
supportedBuilds = {
"u122b4" = { 
    update = "122", build = "04",
    jdk8sha = "", langtoolssha ? "0hhsm23mxvjxmf0jxlhm57s203k88s8xbmk71l8zlnjsz88ni4gx", ...
},
u122b3 = {
    update = "122", build = "03",
    jdk8sha = "", langtoolssha = "", ...
},
}
~~~